samples/cpp/train_HOG.cpp

```diff
@@ -115,214 +100,213 @@ void sample_neg( const vector< Mat > & full_neg_lst, vector< Mat > & neg_lst, co
+    String pos_dir = parser.get<String>("pd");
+    String neg_dir = parser.get<String>("nd");
+    String test_dir = parser.get<String>("td");
+    String obj_det_filename = parser.get<String>("fn");
+    String videofilename = parser.get<String>("tv");
+    int detector_width = parser.get<int>("dw");
+    int detector_height = parser.get<int>("dh");
+    bool test_detector = parser.get<bool>("t");
+    bool train_twice = parser.get<bool>("d");
+    bool visualize = parser.get<bool>("v");
+
+    if (test_detector)
+    {
+        test_trained_detector(obj_det_filename, test_dir, videofilename);
+        return 0;
+    }
 
-                // draw gradient visualization
-                line(visu, Point((int)(x1*zoomFac),(int)(y1*zoomFac)), Point((int)(x2*zoomFac),(int)(y2*zoomFac)), Scalar(0,255,0), 1);
+    if( pos_dir.empty() || neg_dir.empty() )
+    {
+        parser.printMessage();
+        cout << "Wrong number of parameters.\n\n"
+             << "Example command line:\n" << argv[0] << " -pd=/INRIAPerson/96X160H96/Train/pos -nd=/INRIAPerson/neg -td=/INRIAPerson/Test/pos -fn=HOGpedestrian96x160.yml -d\n"
+             << "\nExample command line for testing trained detector:\n" << argv[0] << " -t -dw=96 -dh=160 -fn=HOGpedestrian96x160.yml -td=/INRIAPerson/Test/pos";
+        exit( -1 );
+    }
 
-            } // for (all bins)
+    vector< Mat > pos_lst, full_neg_lst, neg_lst, gradient_lst;
+    vector< int > labels;
 
-        } // for (cellx)
-    } // for (celly)
+    clog << "Positive images are being loaded..." ;
+    load_images( pos_dir, pos_lst, visualize);
+    if (pos_lst.size() > 0)
+    {
+        clog << "...[done]" << endl;
+    }
+    else
+    {
+        clog << "no image in " << pos_dir <<endl;
+        return 1;
+    }
 
+    Size pos_image_size = pos_lst[0].size();
 
-    // don't forget to free memory allocated by helper data structures!
-    for (int y=0; y<cells_in_y_dir; y++)
+    for (size_t i = 0; i < pos_lst.size(); ++i)
     {
-        for (int x=0; x<cells_in_x_dir; x++)
+        if( pos_lst[i].size() != pos_image_size)
         {
-            delete[] gradientStrengths[y][x];
+            cout << "All positive images should be same size!" << endl;
+            exit( 1 );
         }
-        delete[] gradientStrengths[y];
-        delete[] cellUpdateCounter[y];
     }
-    delete[] gradientStrengths;
-    delete[] cellUpdateCounter;
 
-    return visu;
-
-} // get_hogdescriptor_visu
-
-void compute_hog( const vector< Mat > & img_lst, vector< Mat > & gradient_lst, const Size & size )
-{
-    HOGDescriptor hog;
-    hog.winSize = size;
-    Mat gray;
-    vector< Point > location;
-    vector< float > descriptors;
+    pos_image_size = pos_image_size / 8 * 8;
 
-    vector< Mat >::const_iterator img = img_lst.begin();
-    vector< Mat >::const_iterator end = img_lst.end();
-    for( ; img != end ; ++img )
+    if (detector_width*detector_height)
```

Comment:
Please, use `detector_width && detector_height` to avoid warnings from the latest gcc.